utils/build-script

```diff
@@ -774,6 +778,115 @@ class BuildScriptInvocation(object):
 
         return options
 
+    def compute_product_classes(self):
```

Comment:
After reading the rest of the commit, I see what you are doing in terms of product_classes. But still, why not just make it a property called product_classes and then cache the result?